figures/2.figure2/scripts/figure2.r

```diff
@@ -0,0 +1,373 @@
+
+cytokine_values_melted_agg_filtered <- as.matrix(cytokine_values_melted_agg_filtered)
+
+# remove '[NSU]' from the column names
+colnames(cytokine_values_melted_agg_filtered) <- gsub("\\[NSU\\]", "", colnames(cytokine_values_melted_agg_filtered))
+
+col_func <- colorRampPalette(brewer.pal(9, "Purples"))
+# set plot size
+options(repr.plot.width=25, repr.plot.height=5)
+heatmap_anova_cytokines <- (
+    Heatmap(
+        (cytokine_values_melted_agg_filtered),
+        col = brewer.pal(9, "GnBu"),
+        cluster_rows = TRUE,    # Cluster rows
+        cluster_columns = TRUE, # Cluster columns
+        show_row_names = TRUE,  # Show row names
+        show_column_names = TRUE, # Show column names
+        column_names_gp = gpar(fontsize = 12), # Column name label formatting
+        row_names_gp = gpar(fontsize = 12),    # Row name label formatting
+        heatmap_legend_param = list(title = "Abundance", at = c(0, 1)),
+        # make the tiles rectangular
+        rect_gp = gpar(col = NA)
+    )
+)
+
+heatmap_anova_cytokines
+
+# read in the uamp results
+umap_results_path <- file.path("..","..","2.Nomic_nELISA_Analysis/1.umap/PBMC/results/nELISA_plate_430420_umap_PBMC.csv")
+
+umap_results_selected_treatments_path <- file.path("..","..","2.Nomic_nELISA_Analysis/1.umap/PBMC/results/nELISA_plate_430420_umap_PBMC_selected_treatments.csv")
+# read in the data
+umap_results <- read.csv(umap_results_path, header = TRUE, sep = ",")
+umap_results_selected_treatments <- read.csv(umap_results_selected_treatments_path, header = TRUE, sep = ",")
+
+
+umap_results_selected_treatments$oneb_Metadata_Treatment_Dose_Inhibitor_Dose <- factor(umap_results_selected_treatments$oneb_Metadata_Treatment_Dose_Inhibitor_Dose, levels = list_of_treatments)
+
+# set the plot size
+options(repr.plot.width=15, repr.plot.height=5)
+# plot the umap results
+umap_plot_all_treatments <- (
+    ggplot(
+        data = umap_results,
+        aes(
+            x = umap_1,
+            y = umap_2,
+            color = oneb_Metadata_Treatment_Dose_Inhibitor_Dose
+        )
+    )
+    + geom_point()
+)
+umap_plot_all_treatments
+
+# plot the umap results
+umap_plot_selected_treatments <- (
+    ggplot(
+        data = umap_results_selected_treatments,
+        aes(
+            x = umap_1,
+            y = umap_2,
+            color = oneb_Metadata_Treatment_Dose_Inhibitor_Dose
+        )
+    )
+    + geom_point()
+)
+umap_plot_selected_treatments
+
+# show the plots first as assigned names
+# set plot size
+options(repr.plot.width=5, repr.plot.height=5)
+
+# cytokine_scatter_plot
+
+# cytokine_bar_plot
+
+# heatmap_plot_all
+# heatmap_anova_cytokines
+
+# umap_plot_all_treatments
+# umap_plot_selected_treatments
+
+new_heatmap <- as.ggplot(heatmap_anova_cytokines)
+cytokine_scatter_plot_legend <- get_legend(cytokine_scatter_plot)
+umap_plot_selected_treatments_legend <- get_legend(umap_plot_selected_treatments)
+
+
+# remove legends
```

Comment:
Does this mean there is code in here to manually add the legend for the panel A plots? It looks like in the png that the legend is there for the UMAP.

Reply:
Yes, absolutely